src/ActiveDataProvider.php

```diff
@@ -50,7 +51,7 @@
  * For more details and usage information on ActiveDataProvider, see the
  * [guide article on data providers](guide:output-data-providers).
  */
-class ActiveDataProvider
+class ActiveDataProvider extends DataReaderProvider
```

Comment:
How about making it similar to interface?

```suggestion
class ActiveDataReader extends DataReaderProvider
```

Reply:
Fixed